app/controllers/katello/api/v2/repositories_controller.rb

```diff
@@ -267,6 +267,14 @@ def sync
       if params[:source_url].blank? && @repository.url.blank?
         fail HttpErrors::BadRequest, _("attempted to sync without a feed URL")
       end
+      #Hack to support presenters for Pulp3
+      repo = @repository
+      ::Actions::Katello::PulpSelector.send(:define_method, "presenter") do
+        ::Actions::Helpers::Presenter::Delegated.new(self,
+                                          planned_actions(::Actions::Katello::PulpSelector.select([Actions::Pulp::Orchestration::Repository::Sync, Actions::Pulp3::Orchestration::Repository::Sync],
+                                                                                                  repo,
+                                                                                                  SmartProxy.pulp_master)))
+      end
```

Comment:
can you explain why this is here? and not ust defined in pulp_selector?

Reply:
Mostly cause we don't seem to have the repository when looking at presenters. I believe it's happening before the plan..I tried an instance variable as well the input repo[:id] etc.. I might be missing something..